script.js

```diff
@@ -1,12 +1,247 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
+let allShows;
+let allEpisodes;
+
+async function setup() {
+  allShows = getAllShows();
+  const selectedShowId = allShows[0].id; // Selecting  the first show by default
+  allEpisodes = await getEpisodes(selectedShowId);
+
+  createHeader();
+  showSelectField(allShows);
+  episodeSelectField(allEpisodes);
+  addSearchField();
   makePageForEpisodes(allEpisodes);
+  createFooter();
+}
+
+// Fetching episodes for shows
+async function getEpisodes(showId) {
+  try {
+    const response = await fetch(`https://api.tvmaze.com/shows/${showId}/episodes`);
```

Comment:
I would follow the pattern `fetch().then().catch()` to catch errors since it keeps variables better inside the scope they need to exist